apache/superset
# feat: Cross filters scoping

### SUMMARY
Add cross-filter scoping feature (previously removed in PR https://github.com/apache/superset/pull/23216).

In order to set cross filter scoping, user can open the scoping modal either from the configuration menu in upper left corner of the dashboard or from a options menu of any chart that's compatible with cross filters.

User can set a global scoping which will be a default scoping for every chart that supports cross filters. The global scoping can be overriden for each chart individually.

If a custom scoping for a chart is removed, the chart reverts to using the global scoping

Please take note that cross filter will NOT be applied to charts that have datasources incompatible with the emitter chart (target chart uses a different datasource which does not share a common column name with the emitter chart), even if they are in scope of the emitter chart!


### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

https://github.com/apache/superset/assets/15073128/799fd7ed-e9ec-4de5-bf63-598e33c57097



### TESTING INSTRUCTIONS
1. Enable `DASHBOARD_CROSS_FILTERS` feature flag
2. Set global and custom scopings by entering the scoping modal from chart's options menu or from filter's config menu in the upper left corner
3. Apply some cross filterings, verify that they match the scoping set for the source chart

### ADDITIONAL INFORMATION


- [x] Required feature flags: `DASHBOARD_CROSS_FILTERS`
  - [x] Runtime estimates and downtime expectations provided Current: 0.21 s
10+: 0.21 s
100+: 0.14 s
1000+: 0.22 s


CC @kasiazjc @yousoph
